### ISSUE TITLE
use executeSudo for root

### ISSUE THREAD

**Issue by leslie-leder** [int128/gradle-ssh-plugin#180]
Hello,
I do need to run an command as sudo -u root while being connected with lower privileges. 
I've missed somehow the point where to place the root password.
If I use it the proposed way: 

    session(remotes.anyUser) {
      executeSudo '-u root whoami', pty: true
    }
Than I gave the password for anyUser in order to create the session. But where should I place the root password? Something like 

    echo root-passwd | sudo -S -p whoami

wouldn't work since the sudo -S -p is already prepended to the command.
--leslie

**Comment by int128** [comment]
sudo command requires the password of current user (`anyUser` in above) but not the root password.
`executeSudo` automatically gives the password of connected user (`anyUser` in above) to the sudo prompt.

So following script should return `root`.
```groovy
remotes {
  anyUser {
    password = '...'  // for both connection and sudo prompt
  }
}
session(remotes.anyUser) {
  executeSudo 'whoami', pty: true
}
```

**Comment by johnjaylward** [comment]
How can I specify a password for ```executeSudo``` when using an ssh keyfile? Do I specify both the password and the key file in the ```remotes``` configuration?

The host does not allow password auth, only public key, so I don't want to just use the password.

**Comment by int128** [comment]
Sorry my reply is late. `sudoPassword` is available since 2.0.0 Please try it.
